healpix_alchemy/tests/benchmarks/data.py

```diff
@@ -0,0 +1,124 @@
+"""SQLAlchemy sample data for unit tests.
+
+Notes
+-----
+We use the psycopg2 ``copy_from`` rather than SQLAlchemy for fast insertion.
+
+"""
+import io
+
+from astropy.coordinates import SkyCoord, uniform_spherical_random_surface
+from astropy import units as u
+from mocpy import MOC
+import numpy as np
+import pytest
+
+from ...constants import HPX, LEVEL
+from ...types import Tile
+from .models import Galaxy, Field, FieldTile, Skymap, SkymapTile
+
+(
+    RANDOM_GALAXIES_SEED,
+    RANDOM_FIELDS_SEED,
+    RANDOM_SKY_MAP_SEED
+) = np.random.SeedSequence(12345).spawn(3)
+
+
+def get_ztf_footprint_corners():
+    """Return the corner offsets of the ZTF footprint."""
+    x = 6.86 / 2
+    return [-x, +x, +x, -x] * u.deg, [-x, -x, +x, +x] * u.deg
+
+
+def get_footprints_grid(lon, lat, offsets):
+    """Get a grid of footprints for an equatorial-mount telescope.
+
+    Parameters
+    ----------
+    lon : astropy.units.Quantity
+        Longitudes of footprint vertices at the standard pointing.
+        Should be an array of length N.
+    lat : astropy.units.Quantity
+        Latitudes of footprint vertices at the standard pointing.
+        Should be an array of length N.
+    offsets : astropy.coordinates.SkyCoord
+        Pointings for the field grid.
+        Should have length M.
+
+    Returns
+    -------
+    astropy.coordinates.SkyCoord
+        Footprints with dimensions (M, N).
+    """
+    lon = np.repeat(lon[np.newaxis, :], len(offsets), axis=0)
+    lat = np.repeat(lat[np.newaxis, :], len(offsets), axis=0)
+    result = SkyCoord(lon, lat, frame=offsets[:, np.newaxis].skyoffset_frame())
+    return result.icrs
+
+
+def get_random_points(n, seed):
+    with pytest.MonkeyPatch.context() as monkeypatch:
+        monkeypatch.setattr(np, 'random', np.random.default_rng(seed))
+        return uniform_spherical_random_surface(n)
+
+
+def get_random_galaxies(n, cursor):
+    points = SkyCoord(get_random_points(n, RANDOM_GALAXIES_SEED))
+    hpx = HPX.skycoord_to_healpix(points)
+
+    f = io.StringIO('\n'.join(f'{i}' for i in hpx))
+    cursor.copy_from(f, Galaxy.__tablename__, columns=('hpx',))
```

Comment:
Can you explain what is going on with the cursor stuff?

Reply:
Yeah, this is much faster than using SQL. See the docstring.